### PR TITLE
[DOCS] Document escaped backticks for identifiers

### PR DESCRIPTION
**Pull request by jrodewig**
Relates to #62932

### Preview
https://elasticsearch_63079.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/eql-syntax.html#eql-syntax-escape-identifiers

### CODE REVIEW

File: docs/reference/eql/syntax.asciidoc [hunk 1]
```diff
@@ -54,6 +54,30 @@ For example, the following EQL query matches any documents with a
 any where network.protocol == "http"
 ----
 
+[discrete]
+[[eql-syntax-escape-identifiers]]
+==== Escape an event category or field name
+
+Event categories or field names containing non-alphanumeric characters must be
+escaped with enclosing backticks (+++`+++). Non-alphanumeric characters include
+underscores (`_`), dots (`.`), hyphens (`-`), or spaces.
```
Comment (rw-access) [inline docs/reference/eql/syntax.asciidoc:63]: I believe underscores are valid. Also, if a field name starts with a number (i.e. `10x`, then it also needs to be wrapped in backticks)

Comment (jrodewig) [reply docs/reference/eql/syntax.asciidoc:63]: Thanks! I haven't tested this yet so I'll confirm.

Comment (rw-access) [reply docs/reference/eql/syntax.asciidoc:63]: and `dots` are still fine for fields, as long as they are in the middle and not at the end. `process.name` is still valid

Comment (jrodewig) [reply docs/reference/eql/syntax.asciidoc]: Thanks for pointing this out @rw-access. Was able to confirm in my testing. I won't mention dots at all as ES doesn't allow the creation of fields starting or ending with a dot.

**Review by rw-access — APPROVED** [review]

**Review by matriv — COMMENTED** [review]
LGTM

**Pull request merged**